nucypher/blockchain/eth/interfaces.py

```diff
@@ -627,9 +627,13 @@ def send_transaction(self,
                          payload: dict = None,
                          transaction_gas_limit: int = None,
                          confirmations: int = 0,
-                         fire_and_forget: bool = False
+                         fire_and_forget: bool = False  # do not wait for receipt.
                          ) -> dict:
 
+        # TODO: Are confirmations and fire_and_forget conflicting options?
```

Comment:
Or even, should we treat `confirmations == 0` as `fire_and_forget = True`?

Reply:
Good though - `fire_and_forget` is meant for receipts, however.  Suppose you want to wait for a receipt, but not confirmation... or *is* a receipt an implicit single confirmation?

Reply:
Hm, I suppose there is a difference. Even with `confirmations == 0` the call is still waits until the transaction is included into a block. The question is, is there a point in waiting for that but not waiting for confirmations?

Reply:
I think "confirmations" is a bit ambiguous; in our codebase it means how many blocks are on top of the block where you TX is included. So `fire_and_forget` and `confirmations > 0` are indeed incompatible, but that doesn't mean that `(confirmations == 0) == fire_and_forget`: you can wait for the receipt of your TX and as soon as you have one, continue, which is in fact the default behaviour in `nucypher` (i.e., `fire_and_forget=False and confirmations==0`.

To put it logically, `fire_and_forget` implies `confirmations == 0`.

Reply:
:heavy_check_mark: - Let me know if you think the added `ValueError` is enough.